lib/vmpooler/api/v1.rb

```diff
@@ -375,8 +390,16 @@ def atomically_allocate_vms(payload)
       content_type :json
       result = { 'ok' => false }
 
-      if jdata and !jdata.empty?
-        result = atomically_allocate_vms(jdata)
+      if jdata
+        empty = jdata.delete('empty')
+        invalid = jdata.delete('invalid')
+        statsd.increment(statsd_prefix + '.checkout.empty', empty) if !empty.nil?
```

Comment:
`if !` should be replaced with `unless` but you're also calling `#nil?` on those values. In Ruby, all values are truthy except `nil` and `false` so you could simplify 396 and 397 some as:

```ruby
statsd.increment(statsd_prefix + '.checkout.empty', empty) if empty
statsd.increment(statsd_prefix + '.checkout.invalid', invalid) if invalid
```

(I don't know what happens if you try to increment a statsd counter with a nil value, or else I'd say "just lost the `if` entirely")